common/models/user.js

```diff
@@ -421,15 +422,83 @@ module.exports = function(User) {
         return cb(err);
       }
 
-      try {
-        User.validatePassword(newPassword);
-      } catch (err) {
+      this.setPassword(newPassword, options, cb);
+    });
+    return cb.promise;
+  };
+
+  /**
+   * Set this user's password after a password-reset request was made.
+   *
+   * @param {AccessToken} resetToken The token created by resetPassword().
+   * @param {string} newPassword The new password to use.
+   * @param {object} [options]
+   * @callback {Function} callback
+   * @param {Error} err Error object
+   * @promise
+   */
+  User.setPassword = function(userId, newPassword, options, cb) {
+    assert(!!userId, 'resetToken is a required argument');
```

Comment:
It should say `userId` is a required argument.

We should probably test `!!userId || userId ===0` so that `0` can be valid `userId`